pyugrid/grid_io/utils.py

```diff
@@ -1,128 +1,109 @@
-
-    'faces', 'face_face_connectivity', 'face_coordinates_lon', 'face_coordinates_lat'
+    The names_mapping dict has to contain at least: 'nodes_lon', 'nodes_lat'
 
+    Optionally (and mostly required), it can contain: face_face_connectivity',
+    'face_coordinates_lon', 'face_coordinates_lat', and 'faces'
 
     """
+    ug = UGrid()
     attribute_check = {} if attribute_check is None else attribute_check
 
-    # open the file
     nc = netCDF4.Dataset(filename)
 
-    # check if it's the file type we are looking for
-
+    # Check for the specified attributes.
     for name, value in attribute_check:
         if nc.getncattr(name).lower() != value:
             raise ValueError('This does not appear to be a valid file:\n'
                              'It does not have the "{}"="{}"'
                              'global attribute set'.format(name, value))
-    # create an empty UGrid:
-    ug = pyugrid.UGrid()
 
-    # load the nodes -- this is required
-    # nodes are usually stored in two different arrays
+    # Nodes.
     lon = nc.variables[names_mapping['nodes_lon']]
     lat = nc.variables[names_mapping['nodes_lat']]
 
-    num_nodes = lon.shape[0]
+    num_nodes = lon.size
     ug.nodes = np.zeros((num_nodes, 2), dtype=lon.dtype)
     ug.nodes[:, 0] = lon[:]
     ug.nodes[:, 1] = lat[:]
 
-    # load the faces
+    # Faces.
     faces = nc.variables[names_mapping['faces']]
-    # does it need to be transposed?
-    # assume there are more than three triangles...
+    # FIXME: This logic assumes there are more than three triangles.
     if faces.shape[0] <= faces.shape[1]:
-        # Fortran order -- needs to be transposed
+        # Fortran order.
         faces = faces[:].T
     else:
         faces = faces[:]
-    # is it one-indexed?
+
+    # One-indexed?
     if faces.min() == 1:
         one_indexed = True
-        faces -= 1
-        ug.faces = faces
     else:
         one_indexed = False
 
-    # load the connectivity array: optional
+    if one_indexed:
+        faces -= 1
+    ug.faces = faces
+
+    # Connectivity (optional).
     if 'face_face_connectivity' in names_mapping:
-        face_face_connectivity = nc.variables[names_mapping['face_face_connectivity']]
-        # does it need to be transposed?
-        # assume there are more than three triangles...
+        face_face_connectivity = nc.variables[names_mapping['face_face_connectivity']]  # noqa
+        # FIXME: This logic assumes there are more than three triangles.
         if face_face_connectivity.shape[0] <= face_face_connectivity.shape[1]:
-            # Fortran order -- needs to be transposed
+            # Fortran order.
             face_face_connectivity = face_face_connectivity[:].T
         else:
             face_face_connectivity = face_face_connectivity[:]
         if one_indexed:
             face_face_connectivity -= 1
-        ug.face_face_connectivity = face_face_connectivity[:]
+        ug.face_face_connectivity = face_face_connectivity
```

Comment:
Actually that should not be there (my bad).  Even though I vote for *lazy*, that is inconsistent and the transposed version is an array and the non-transposed a netCDF4 variable.  I will fix that in another PR.